src/components/App/App.js

```diff
@@ -40,17 +51,141 @@ function App() {
     setSignUpPopupOpen(false);
     setConfirmPopupOpen(false);
   }
-  function handelSignUpSubmit(username) {
-    setCurrentUser({ username: username, isLoggedIn: false });
-    setSignUpPopupOpen(false);
-    setConfirmPopupOpen(true);
-  }
-  function handelSignInSubmit(username) {
-    setCurrentUser({ username: username, isLoggedIn: true });
-    setSignInPopupOpen(false);
+  useEffect(() => {
+    const token = localStorage.getItem('jwt');
+    if (token) {
+      checkToken(token)
+        .then((res) => {
+          console.log('Username set, ', res.name);
+          setCurrentName(res.name);
+          setLoggedIn(true);
+        })
+        .catch((err) => console.log(`Error: ${err}`));
+    }
+  }, []);
+
+  useEffect(() => {
+    loggedIn &&
+      getUserData()
+        .then((res) => setCurrentUser(res))
+        .catch((err) => console.log(`Error: ${err}`));
+  }, [loggedIn]);
+
+  function handleSearchSubmit(search) {
+    setIsSearching(true);
+    newsProxyApi.getNews(search).then((cards) => {
+      if (!cards.articles || cards.articles.length === 0) {
+        setIsNotFound(true);
+        setIsSearched(false);
+        setIsSearching(false);
+      } else {
+        setCards(cards.articles);
+        localStorage.setItem('searchResults', JSON.stringify(cards.articles));
+        setKeyword(search);
+        localStorage.setItem('keyword', search);
+        console.log(keyword);
+        setIsSearched(true);
+        setIsSearching(false);
+        setIsNotFound(false);
+      }
+    });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**